ze/tracer_ze_helpers.include.c

```diff
@@ -325,27 +371,60 @@ static struct _ze_event_h * _get_profiling_event(
 
 static void _profile_event_results(ze_event_handle_t event);
 
-static inline void _unregister_ze_event(ze_event_handle_t event, int remove_cl, int get_results) {
+static inline void _on_created_event(ze_event_handle_t event) {
+#ifdef THAPI_DEBUG
+  struct _ze_obj_h *o_h = NULL;
+  FIND_ZE_OBJ(&event, o_h);
+  if (o_h) {
+    THAPI_DBGLOG("Event already registered: %p", event);
+    return;
+  }
+
+  intptr_t mem = (intptr_t)calloc(1, sizeof(struct _ze_obj_h));
+
+  if (mem == 0) {
+    THAPI_DBGLOG_NO_ARGS("Failed to allocate memory");
+    return;
+  }
+
+  o_h = (struct _ze_obj_h *)mem;
+  o_h->ptr = (void *)event;
+  o_h->type = EVENT;
+
+  ADD_ZE_OBJ(o_h);
+#else
+  (void)event;
+#endif
+}
+
+static inline void _on_destroy_event(ze_event_handle_t event) {
   struct _ze_event_h *ze_event = NULL;
 
+#ifdef THAPI_DEBUG
+  struct _ze_obj_h *o_h = NULL;
+  FIND_AND_DEL_ZE_OBJ(&event, o_h);
+  if (!o_h) {
+    THAPI_DBGLOG("Could not find event: %p", event);
+  }
+#endif
+
   FIND_AND_DEL_ZE_EVENT(&event, ze_event);
-  if (!ze_event)
+  if (!ze_event) {
```

Comment:
Should we print a warning?

Reply:
It is not really an error to destroy an event that was already profiled or reset or not used. The vent not being in the event to trace pool just means we should not ask for profiling infos.

This is different in the reset case where if the user resets the event, it should have triggered before and we should have profiled it (It may still trigger on user events that are not used for profiling, but forhost synchronization)